app/app_abci.go

```diff
@@ -29,7 +29,8 @@ func (app *OKExChainApp) DeliverTx(req abci.RequestDeliverTx) (res abci.Response
 		tx, err := evm.TxDecoder(app.marshal)(req.Tx)
 		if err == nil {
 			//optimize get tx gas price can not get value from verifySign method
-			app.blockGasPrice = append(app.blockGasPrice, tx.GetGasPrice())
+			app.gpo.CurrentBlockGPs.AddGP(tx.GetGasPrice())
+			app.gpo.CurrentBlockGPs.AddGas(tx.GetGas())
```

Comment:
tx.GetGas is set by the user and cannot get the real gasUsed. I suggest that the response of DeliverTx `resp := app.BaseApp.DeliverTx(req)` should be used here

Reply:
down